onnx/backend/test/case/node/dropout.py

```diff
@@ -11,19 +11,14 @@
 from . import expect
 from onnx import helper
 
+def dropout(X, drop_probability=0.5, seed=0):  # type: ignore
```

Comment:
This PR seems to also change the op specification about the default value of ratio/drop_probability. The current spec says that default value is 0. Two questions: (a) Why change the default-value? I understand that it makes it more compatible with PyTorch/TF, but I wonder why it was made 0 in the first-place and what things would break if we change it thus, (b) If we must change the default-value, the spec also needs to be changed/updated.

Reply:
@gramalingam The rationale is to be compatible with PyTorch/TF and even ORT has default as 0.5. Happy to update the spec but lets first understand why the default of zero was chosen, maybe @wschin might know?

Reply:
@gramalingam good catch!

I don't suggest to change this default value. I believe the "drop elimination" optimizer was assuming that the default value is 0 so that in inference case it's just a copy and can be removed.

Reply:
Thanks, @linkerzhang then in that case we need to change the "default_dropout" test to pass the probability of 0.5 because on ORT side default probability is set to 0.5 (same as PyTorch/TF). In anycase the reference implementation is not doing the scaling as it is done in PyTorch/TF and ORT.

Reply:
@codemzs : it sounds like ORT implementation does not match the spec then. Either the spec needs to be changed or the ORT implementation needs to change (I don't think it is just a matter of changing the test-case).

Reply:
I suggest not updating the spec, which costs more than the benefits it brings.

Reply:
@linkerzhang I have made a note of your suggestion but the final decision will be taken by the team on Monday morning, this includes @ebarsoum , @gramalingam , @wschin , @yuanbyu , @SherlockNoMad , @prasanthpul , etc.

Reply:
Thanks @linkerzhang ! lets see what other reviewers have to say? @ebarsoum @yuanbyu @wschin @gramalingam

Reply:
@linkerzhang Please take a look again, we added a training_mode input to Dropout so that existing inference behavior does not break while the operator can support training that can be dynamically switched with default probability of 0.5